services/vision/mlvision/detector.go

```diff
@@ -59,17 +59,26 @@ func attemptToBuildDetector(mlm mlmodel.Service) (objectdetection.Detector, erro
 			return nil, err
 		}
 
-		locations := unpack(outMap, "location")
-		if len(locations) == 0 {
-			locations = unpack(outMap, DefaultOutTensorName+"0")
+		locations, err := unpack(outMap, "location")
+		if err != nil || len(locations) == 0 {
+			locations, err = unpack(outMap, DefaultOutTensorName+"0")
+			if err != nil {
+				return nil, err
```

Comment:
Rather than overwrite err, can you combine the two errors? You can use multierr.Combine
https://pkg.go.dev/go.uber.org/multierr#section-readme
you will need to define err2 outside of the scope of the if statement in order for "locations" to be correctly filled, too.